src/io.jl

```diff
@@ -76,14 +78,19 @@ function read_datafile(filename::String)
 
     # Extract data arrays
     n_nodes = polydeg + 1
-    data = Array{Float64}(undef, n_nodes, n_nodes, n_elements, n_variables)
+
+    if ndims == 3
+      # Read 3d data
+      data = Array{Float64}(undef, n_nodes, n_nodes, n_nodes, n_elements, n_variables)
+    else
+      # Read 2d data
+      data = Array{Float64}(undef, n_nodes, n_nodes, n_elements, n_variables)
+    end
```

Comment:
Any ideas how to make this a one-liner?

Reply:
I think this is quite clear. You could/should also check whether `ndims == 2` in the second case and throw an error in an `else` block, I think.